zng/docs/zson.md

```diff
@@ -236,9 +228,23 @@ Internal type names are represented by integers while external names are
 represented by identifiers.  When ZSON data is organized into a sequence
 comprised of two or more subsequences where each subsequence has its own
 type context, the internal names may be "reused" across type contexts to refer
-to different types but external names must have the same type value across
-subsequences.  It is a "type mismatch" error if an external name has different
-type values across subsequences.
+to different types but external names should generally have the same type value
+throughout.  That said, external type definitions _may_ vary and an implementation
+must properly handle changing external name by updating the binding between
+the type name and the new type for all subsequent values in the sequence.
+
+> Handling type definition conflicts outside of the scope of ZSON, but an implementation
+> may police all external definitions so that "type conflicts" are rejected with
+> an error, or it may arrange to name types with a version number, e.g., example.0,
+> example.1, etc then have operators that understand that example.* are all variations
```

Comment:
```suggestion
> an error, or it may arrange to name types with a version number (e.g., example.0,
> example.1, etc.), and then have operators that understand that example.* are all variations
```